dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/LogStrategyTest.kt

```diff
@@ -53,8 +56,8 @@ import org.mockito.junit.jupiter.MockitoSettings
 @MockitoSettings()
 internal abstract class LogStrategyTest {
```

Comment:
I'm thinking maybe we could create an abstract "PersistingStrategy" class as a generic version of LogStrategy. That way the LogStrategy itself would only be passing the LogSerializer, and testing that class would mean that you don't care how to serialize (and deserialize for the test) logs, and just care that data is written and read exactly as is. SRP :)

Reply:
Yup, I see your point and I think we should do this but I would rather do this in another PR. This one is getting way to big and we already moved lots of things in it. I think is better to merge it as it is (we already extracted the serialisation part from the LogFileWriter) and then you can merge your UserInfo PR because you will have conflicts there (you need to move all the serialisation part in the LogSerializer) . We clean this part and then it will be easy to switch and clean also the Strategy.

Reply:
Actually let me just remove the generic part from the logger and move the serializer in the LogStrategy, it doesn't make sense to keep it there actually.

Reply:
Sorry, I deleted by mistake the previous comment. I agree actually it will be a good next step. I would rather do this in a next PR as this one is getting too big and you already have conflicts on the UserInfo ticket. I will handle the previous comments and then let's merge this asap as we already moved lots of things and then I will assist if you want to solve the conflicts on your PR.